thread/test/perf_workpool.cpp

```diff
@@ -30,9 +30,22 @@ DEFINE_uint64(fires, 80000, "How many tasks to fire");
 static photon::WorkPool* pool;
 static std::atomic<uint64_t> sum_time;
 
-void task_async() {
+__attribute__ ((noinline))
+static void do_fill(char* buf, size_t size) {
+    memset(buf, 0, size);
+}
+// A function that costs approximately 1 us
+static void func_1us() {
+    for (int i = 0; i < 5; ++i) {
+        char buf[32 * 1024];
+        do_fill(buf, sizeof(buf));
+    }
+}
+
+static void task_async() {
     photon::semaphore sem(0);
     for (uint64_t i = 0; i < FLAGS_fires; ++i) {
+        func_1us();
```

Comment:
But what's the purpose to permanently add this wait?

Reply:
To simulate real work loads, and make the test results more reasonable.
Otherwise the performance of case 1 is too low.